lib/ball.js

```diff
@@ -0,0 +1,61 @@
+function Ball(options, canvas){
+  this.canvas = canvas;
+  this.ctx = this.canvas.getContext('2d');
+  this.r = options.r > 0 && options.r < 20 && options.r || 8; // can be fixed all the time?
```

Comment:
You might want to move this out to a function and/or private method. It's kind of confusing to figure out what this code does. Also, it might be worth looking at default arguments in ES6.